pkg/cmd/accept/exec.go

```diff
@@ -94,10 +94,24 @@ func (o *Options) runWithClient(kubeClient *kubernetes.Clientset, clusterClient
 }
 
 func (o *Options) accept(kubeClient *kubernetes.Clientset, clusterClient *clusterclientset.Clientset, clusterName string, waitMode bool) (bool, error) {
-	approved, err := o.approveCSR(kubeClient, clusterName, waitMode)
+	managedCluster, err := clusterClient.ClusterV1().ManagedClusters().Get(context.TODO(),
+		clusterName,
+		metav1.GetOptions{})
 	if err != nil {
-		return approved, fmt.Errorf("fail to approve the csr for cluster %s: %v", clusterName, err)
+		return false, fmt.Errorf("fail to get managed cluster")
```

Comment:
```suggestion
		return false, fmt.Errorf("fail to get managedcluster %s: %v", clusterName, err")
```
more consistent with the previous error message.

Reply:
Yes, this is part of our design. This will work for only EKS based registration. If somebody is using default csr authentication, and they add these annotations, their registration won't complete. If they want to complete the registration with csr based authentication, they have to remove this annotation.